augmentedreality/Common/source/Android/ArCoreWrapper.cpp

```diff
@@ -289,8 +289,7 @@ void ArCoreWrapper::createArSession()
 
   if (qApp->checkPermission(cameraPermission) == Qt::PermissionStatus::Undetermined)
   {
-    qApp->requestPermission(cameraPermission, this, [this, &loop]() 
-    {
+    qApp->requestPermission(cameraPermission, [&loop]() {
```

Comment:
I guess it's necessary to have a context object in second position https://doc.qt.io/qt-6/qcoreapplication.html#requestPermission-1

`QEventLoop` is a `QObject`, it makes sens to use it as context.

```cpp
qApp->requestPermission(cameraPermission, &loop, [&loop]() {
```

Reply:
It looks like the call without a context worked as well

qcoreapplication.h:165
```
    // requestPermission to a functor or function pointer (without context)
    template <typename Func, std::enable_if_t<
        !QtPrivate::FunctionPointer<Func>::IsPointerToMemberFunction
        && !std::is_same<const char *, Func>::value, bool> = true>
    void requestPermission(const QPermission &permission, Func func)
    {
        requestPermission(permission, nullptr, std::move(func));
    }
```

Reply:
Nevermind, I was looking at the dev branch my mistake

Reply:
It was a trap :)